### DISCUSSION TITLE
ipfs pinning

### DISCUSSION THREAD

**Discussion by unicomp21** [snapshot-labs/snapshot#3436] [General]
I've got gigabytes of data and need ipfs p2p replication between browsers to cut down on aws egress charges.  I've tried the ipfs node in brave browser, but can't make it work.  Could anyone give me some tips?

**Comment by ChaituVR** [comment]
Hi @unicomp21 Can share more details about what you are trying to do, can share the URL that is not working?

**Comment by unicomp21** [comment]
I'm trying to create a webgl experience, think of it as a game w/ a "lot" of assets?  Gigabytes of textures, meshes, etc.  ie Lots of gltf files.  My intention is to monetize via brave rewards for creators, but if my dataset is distributed via aws, the egress fees will likely exceed what I make in brave rewards?  So, I need a way for the browsers to replicate my data using ipfs p2p replication and the dht?  Having users manually enable the ipfs node within brave browser is not a problem, I can do that.

Does all this make sense @ChaituVR ?

Here is a link to a small test project I'm trying to get going:
https://ipfs.io/ipfs/QmPGbMufnBNm5CocEdi8CKvVHEz1B1RMgFtuWWrh7AhpDg?filename=index.html

And here is the ipns publish to the above link:
https://ipfs.io/ipns/k51qzi5uqu5dhxuru75q23pcmg8rzdmdiqsnwqtqdowb1z4re25uomrz8ntxir

The links above were generated using the ipfs node bundled w/ brave browser.

Would filebase give me the p2p replication I'm looking for?  Or am I going about this all wrong?

**Comment by unicomp21** [comment]
I currently get 504's on the above ipfs url's when trying to hit them from the brave node used to publish them.

**Comment by unicomp21** [comment]
cc: @BrendanEich

**Comment by ChaituVR** [comment]
How is this related to snapshot?

**Comment by unicomp21** [comment]
It's not, just one dev begging another for advice on how best to use ipfs.

**Reply by ChaituVR** [reply]
Sorry, I was thinking it was a snapshot related issue. 😅 So these links are not working because maybe these IPFS files are not pinned? you have to use a third-party pinning service like:
[pinata](https://www.pinata.cloud/)
[fleek](https://fleek.co/), 
https://web3.storage/ 
https://www.infura.io/product/ipfs 
https://www.4everland.org/

At snapshot, we pin all data to all the above services based on their availability, refer to https://github.com/snapshot-labs/pineapple

**Reply by unicomp21** [reply]
Thanks!

**Comment by unicomp21** [comment]
Just to make sure I'm clear.  If I have my users run the ipfs node bundled w/ brave browser, this will cut down on the egress charges for the ipfs pinning service I choose to use, correct?  ie I don't want to be paying aws egress charges for each user to download the gigabytes in the application dataset.
